### ISSUE TITLE
auto naming of objects breaking python

### ISSUE THREAD

**Issue by iscai-msft** [Azure/autorest#3750]
In the cases of auto-naming, it appears that `-` is getting tacked on (i.e. in this case, the object schema for the `sku`. property in `AvailableServiceSku` [here](https://github.com/00Kai0/azure-rest-api-specs/blob/kaihuis-track2-datamigration/specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2018-07-15-preview/definitions/Services.json#L29), we get `AvailableServiceSku-Sku`). This causes Python generated. code to not work (original issue [here](https://github.com/Azure/autorest.python/issues/854)). I remember talking about not having m4 name classes and forcing users to specify a name in the swagger, but since it looks like we're currently still supporting this behavior, I think it's best to change the autorest-generated name in the autorest repo, since other languages run into problems here too (I believe js classes can't have `-` either, maybe some others). 

The service team isn't blocked on this rn bc of `x-ms-client-name`

cc @00Kai0

**Comment by timotheeguerin** [comment]
Looking at this I think this is a side effect of my fix to prevent renaming to already used values. 
https://github.com/Azure/autorest.modelerfour/pull/374

This is not a new issue as the problem here would have been that it was Converting `AvailableServiceSku-Sku` to `AvailableServiceSku` which already exist. But now as it find it already exists it won't rename it.

**Comment by timotheeguerin** [comment]
I have a fix for that, wondering now what should be the correct behavior. 
Right now I have it that it will try to remove duplicate consecutive words and check if the new name already exists. 
If it does it will try to style the name without removing duplicates and check if that new name already exists.
If it doesn't use that otherwise just ignore the rename. 

Now the problem with this is if there is something that would resolve into a duplicate(Having user explicitly name 2 model FooBar and Foo-Bar) it will not rename `Foo-Bar` and ignore it which will then cause issue in the generated sdk. Question is should we still fail here saying there is a duplicate?

**Comment by iscai-msft** [comment]
sorry i totally forgot to respond on friday.

I briefly remember the discussion to remove duplicates in the name (can't fully remember the specifics). But in this case (where removing the duplicates clashes with an existing name), I'm thinking it would be better to allow them. I think it's safest to stick to alphanumeric where possible, and I think having something like `AvailableServiceSkuSku` is better than `AvailableServiceSku2` (though both aren't great lmao).

Another option is to raise on things like this, which from a stylistic standpoint seems good, since neither of these names seem like good names in an SDK. This might be toeing the line for "breaking changes" though

I would say for that last example: if 2 names resolve to the same name (i.e., I would hope both would resolve to `FooBar`), we should raise an error. I don't think we want a generated code that has all of these very similar class / method names, with just a `-` etc thrown in to differentiate.

@daviwil do you have any thoughts?
